### PR TITLE
fix: Generated schemas contains a valid meta-schema URI reference

### LINKED ISSUE #2519: Generated schemas contain a deprecated meta-schema URI reference
## Description

[JSON schemas](https://json-schema.org/) generated by the client contain an unversioned meta-schema URI reference.

image

The [value](https://json-schema.org/understanding-json-schema/reference/schema.html) referenced here:
https://github.com/fabric8io/kubernetes-client/blob/e618795e882a7d3d154bfb4f8513ebdf5378729e/kubernetes-model-generator/pkg/schemagen/generate.go#L249

should be updated to the applicable [version](https://json-schema.org/specification-links.html) (e.g. `https://json-schema.org/draft/2019-09/schema`).

The generated schema files `kube-schema.json`, `kube-validation-schema.json` and `validation-schema.json` in each model module should contain the updated entries.

See also:
- https://github.com/eclipse/jkube/issues/415#issuecomment-701209519
- https://json-schema.org/understanding-json-schema/reference/schema.html

### PR DESCRIPTION
**Pull request by manusa**
## Description
fix #2519: Generated schemas contains a valid meta-schema URI reference (`http://json-schema.org/draft-05/schema#`)

 - Improved model generation script (peforms gobuild by default > no mvn)
 - Fixed problem with kubernetes-model-events generation
 - Fixed problem with chaosmesh model generation (invalid go module indirect reference)
 - Removed no longer generated/maintained kube-validation-schema.json
 - Added 5.x Migration guide

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - ~~[ ] I have implemented unit tests to cover my changes~~
 - [x] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
@rohanKanojia  please take a good look because some models/schemas have changed more than expected but they shouldn't have (e.g. admissionregistration). This is really awkward since their `generate.go` files and dependencies haven't changed.

**Review by oscerd — APPROVED** [review]

**Comment by rohanKanojia** [comment]
strange, I've gone over your PR and everything seems okay to me code wise. We also have tests for `ValidatingWebhookConfiguration`, `MutatingWebhookConfiguration`, `SubjectAccessReview` if there was something wrong they would have failed.

File: doc/MIGRATION-v5.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Migration from 4.x to 5.x
+
+## kube-validation-schema.json files renamed to validation-schema.json
```
Comment (rohanKanojia) [inline doc/MIGRATION-v5.md:3]: Why this migration guide doesn't provide information about `Doneable` being removed and change in `edit()` signatures? Or does this PR only document it from the model perspective?

**Review by rohanKanojia — APPROVED** [review]
:+1:

**Comment by manusa** [comment]
[merge]

**Comment by manusa** [comment]
[merge]

**Pull request merged**